### PR TITLE
Fix the use of Observables and `nan_color` in `rplot`

### LINKED ISSUE #401: Rplot fails with nancolor gl_convert
I try to plot a Raster object with the new rplot function, and this fails with the error below. The figure and the Axis are correctly constructed, but the data is not shown. Even though there is data available. A failing nc file is attached to the issue. 

I can circumvent the problem by giving the nancolor keyword as a Colorant type, so that it does not need to do the gl_convert from a symbol. 

```julia
using Rasters

julia> s = Raster("smallrqa.nc")
[ Info: No `name` or `key` keyword provided, using first valid layer with name `:layer`
100×100 Raster{Union{Missing, Float32},2} layer with dimensions: 
  X Mapped{Float64} Float64[4.8e6, 4.80002e6, …, 4.80196e6, 4.80198e6] ForwardOrdered Regular Points crs: EPSG mappedcrs: EPSG,
  Y Mapped{Float64} Float64[2.1e6, 2.09998e6, …, 2.09804e6, 2.09802e6] ReverseOrdered Regular Points crs: EPSG mappedcrs: EPSG
extent: Extent(X = (4.8e6, 4.80198e6), Y = (2.09802e6, 2.1e6))
missingval: missing
crs: EPSG:4326
mappedcrs: EPSG:4326
values:              2.1e6       2.09998e6   2.09996e6   2.09994e6     2.09992e6    2.0999e6   …   2.09812e6   2.0981e6    2.09808e6    2.09806e6   2.09804e6     2.09802e6
 4.8e6       0.105774    0.35083     0.295654    0.116394      0.00674057  -0.118286      -0.138672   -0.0878296  -0.00332451   0.0217133  -0.000139832   0.0240936
 4.80002e6   0.0339966   0.255127    0.228516    0.232544      0.10553      0.0705566     -0.0492859  -0.0497437  -0.103455    -0.109131   -0.116211     -0.0203247
 4.80004e6  -0.0279999   0.172363    0.341797    0.415283      0.19397     -0.0179138      0.0262451   0.0397644  -0.0820923    0.0128403  -0.0457764    -0.104858
 ⋮                                                             ⋮                       ⋱                           ⋮                                     
 4.80192e6  -0.15979    -0.1427     -0.187988   -0.079895      0.0490112    0.0987549  …  -0.117554   -0.112      -0.118896    -0.0474243   0.00404358   -0.0430298
 4.80194e6  -0.0429382  -0.119324   -0.161377   -0.0843506    -0.0557251   -0.0262146     -0.0050087  -0.144287   -0.156738    -0.0388794   0.0173645     0.0650024
 4.80196e6  -0.22876    -0.26123    -0.11615     0.000240088  -0.09375     -0.34082       -0.0184326  -0.128662   -0.069519    -0.0498657  -0.027298      0.0342407
 4.80198e6  -0.32666    -0.238037   -0.107788   -0.00111103   -0.197021    -0.430176      -0.170532   -0.113953   -0.100647    -0.0834351  -0.067627     -0.0272827

julia> Rasters.rplot(s)

```

__I have pasted a _complete_ error from function call down to the end__
```julia
julia> Rasters.rplot(small)
┌ Warning: Can't convert Observable{Symbol} to opengl, for attribute nancolor
└ @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:427
Error showing value of type Makie.FigureAxisPlot:
ERROR: MethodError: no method matching gl_convert(::Symbol)

Closest candidates are:
  gl_convert(::Quaternion)
   @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/glshaders/particles.jl:28
  gl_convert(::Observable{T}) where T<:GeometryBasics.Mesh
   @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:197
  gl_convert(::Observable{T}) where T<:Union{Float32, Float64, Int32, UInt32, GeometryBasics.ZeroIndex{Int32}, GLMakie.GLAbstraction.GLProgram, GLMakie.GLAbstraction.Shader, GLMakie.GLAbstraction.GPUArray, StaticArraysCore.StaticArray{Tuple{N}, T, 1} where {N, T}, GeometryBasics.GLIndex, StaticArraysCore.SMatrix}
   @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:227
  ...

Stacktrace:
  [1] #map#13
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:564 [inlined]
  [2] map
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:562 [inlined]
  [3] const_lift
    @ ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUtils.jl:106 [inlined]
  [4] gl_convert(s::Observable{Symbol})
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLUniforms.jl:228
  [5] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window, context::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:425
  [6] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/N16Fq/src/GLAbstraction/GLTypes.jl:399
  [7] assemble_shader(data::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/glshaders/visualize_interface.jl:108
  [8] draw_heatmap(screen::Any, main::Any, data::Dict)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/glshaders/image_like.jl:58
  [9] (::GLMakie.var"#222#227"{GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}})(gl_attributes::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:434
 [10] (::GLMakie.var"#168#171"{GLMakie.var"#222#227"{GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}}, GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}})()
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:111
 [11] get!(default::GLMakie.var"#168#171"{GLMakie.var"#222#227"{GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}}, GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}}, h::Dict{UInt64, GLMakie.GLAbstraction.RenderObject}, key::UInt64)
    @ Base ./dict.jl:468
 [12] cached_robj!(robj_func::GLMakie.var"#222#227"{GLMakie.Screen{GLFW.Window}, Scene, Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}}}, screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:84
 [13] draw_atomic
    @ ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:393 [inlined]
 [14] insert!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::Heatmap{Tuple{Vector{Float64}, Vector{Float64}, Matrix{Float32}}})
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/drawing_primitives.jl:126
 [15] insertplots!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:435
 [16] insertplots!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene) (repeats 2 times)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:438
 [17] display_scene!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:374
 [18] GLMakie.Screen(scene::Scene, config::GLMakie.ScreenConfig; visible::Nothing, start_renderloop::Bool)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:391
 [19] GLMakie.Screen(scene::Scene, config::GLMakie.ScreenConfig)
    @ GLMakie ~/.julia/packages/GLMakie/N16Fq/src/screen.jl:387
 [20] getscreen(::Module, ::Scene; screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:336
 [21] getscreen
    @ ~/.julia/packages/Makie/gAmAB/src/display.jl:324 [inlined]
 [22] display(figlike::Makie.FigureAxisPlot; backend::Module, update::Bool, screen_config::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:148
 [23] display(figlike::Makie.FigureAxisPlot)
    @ Makie ~/.julia/packages/Makie/gAmAB/src/display.jl:133
 [24] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [25] invokelatest
    @ ./essentials.jl:813 [inlined]
 [26] print_response(errio::IO, response::Any, show_value::Bool, have_color::Bool, specialdisplay::Union{Nothing, AbstractDisplay})
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:303
 [27] (::REPL.var"#57#58"{REPL.LineEditREPL, Pair{Any, Bool}, Bool, Bool})(io::Any)
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:285
 [28] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:555
 [29] print_response(repl::REPL.AbstractREPL, response::Any, show_value::Bool, have_color::Bool)
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:283
 [30] (::REPL.var"#do_respond#80"{Bool, Bool, REPL.var"#93#103"{REPL.LineEditREPL, REPL.REPLHistoryProvider}, REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::Any, ok::Bool)
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:897
 [31] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [32] invokelatest
    @ ./essentials.jl:813 [inlined]
 [33] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2647
 [34] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/.julia/juliaup/julia-1.9.0-rc1+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1298
 [35] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:514
```
image

[smallrqa.zip](https://github.com/rafaqz/Rasters.jl/files/11050656/smallrqa.zip)

### PR DESCRIPTION
**Pull request by asinghvi17**
Fixes #401

### CODE REVIEW

**Review by rafaqz — APPROVED** [review]

File: ext/RastersMakie.jl [hunk 1]
```diff
@@ -100,7 +100,7 @@ function Rasters.rplot(position::GridPosition, raster::Union{AbstractRaster{T,2,
             title, xlabel, ylabel 
         )
         # plot to the axis with the specified plot type
-        plot = plot!(plottype, axis, raster; colormap, colorrange, nancolor)
+        plot = plot!(plottype, axis, raster; colormap, colorrange, nan_color = lift(to_color, convert(Observable, nancolor))
```
Comment (asinghvi17) [inline ext/RastersMakie.jl:103]:
```suggestion
        plot = plot!(plottype, axis, raster; colormap, colorrange, nan_color = lift(to_color, convert(Observable, nancolor)))
```
missed parenthesis

**Comment by rafaqz** [comment]
Is there a way to test this?

**Comment by asinghvi17** [comment]
I've tried a basic test by adding a simple animation to the docs, hopefully it doesn't take too long to render!

**Comment by rafaqz** [comment]
I think its nice to have an animation in the docs anyway :)

**Comment by asinghvi17** [comment]
Yep should be good!  I'm deliberately not putting any fancy surface or mesh stuff here, since CairoMakie is pretty slow at rendering those, but standard heatmaps work pretty fast.

**Comment by asinghvi17** [comment]
Good news: fixed the problems locally so will push soon.
Uploading rplot.mp4…



Bad news: I tried to solve the problem of colorbar labels clipping into neighboring axes, and I don't think it's possible with the way Makie layouts currently work.  I'll probably comment that patch out for later :(

download-1

**Comment by asinghvi17** [comment]
Also, for posterity, here's the fancy surface/heatmap Raster animation that I posted on Slack:

https://user-images.githubusercontent.com/32143268/227770232-8733b0c1-638f-49b2-97f0-c78b0bbd0e2f.mp4

**Comment by rafaqz** [comment]
I would usually fix the colour bounds for an animation so you can compare the change in each frame.

But can we even do that fir a stack where eaxh layer needs different limits?

**Comment by asinghvi17** [comment]
Yeah...to be entirely honest, I'm not too sure why that's happening either.  It could be fixed to the limits of the first "frame" (initial input) but that has its own set of issues.

Of course, you can always pass a vector of preconstructed color ranges.  This could also be an interesting thing to do in Makie, where there's an option to have the color range be cumulative across all updates.

**Comment by asinghvi17** [comment]
This should be good to merge!

**Pull request merged**